rescript-relay-documentation/docs/tutorial/arrays-lists.md

```diff
@@ -0,0 +1,139 @@
+---
+id: tutorial-arrays-lists
+title: Arrays and Lists
+sidebar_label: Arrays and Lists
+---
+
+:::info
+This tutorial is forked from the [official Relay tutorial](https://relay.dev/docs/tutorial/intro/), and adapted to RescriptRelay. All the credit goes to the Relay team for writing the tutorial.
+:::
+
+# Arrays and Lists
+
+So far we’ve only dealt with components that have a single instance of the components they’re composed from. For example, we’re only showing a single Newsfeed story, and within that story there’s just a single author with a single profile picture. Let’s look at how to handle more than one of something.
+
+GraphQL includes support for arrays, which in GraphQL are called _lists._ A field can be not only a single scalar value but an array of them, or not only a single edge but an array of edges. The schema specifies whether each field is a list or not, but, oddly, the GraphQL query syntax doesn’t distinguish between selecting a singular field and selecting a list — a quirky exception to the design principle that a GraphQL response should have the same shape as the query.
+
+Request:
+
+```
+query MyQuery {
+  viewer {
+    contacts { // List of edges
+      id // field on a single item
+      name
+    }
+  }
+}
+```
+
+Response:
+
+```
+{
+  viewer: {
+    contacts: [ // array in response
+      {
+        id: "123",
+        name: "Chris",
+      },
+      {
+        id: "789",
+        name: "Sue",
+      }
+    ]
+  }
+}
+```
+
+As it happens, the schema in our example app has a `topStories` field that returns a list of Stories, as opposed to the `topStory` field we're currently using that returns just one.
+
+To show multiple stories on our newsfeed, we just need to modify `Newsfeed.res` to use `topStories`.
+
+### Step 1 — Select a list in the fragment
+
+Open `Newsfeed.res` and find `NewsfeedQuery`. Replace `topStory` with `topStories`.
+
+```rescript
+const NewsfeedQuery = graphql`
+  query NewsfeedQuery {
+    // change-line
+    topStories {
+      ...StoryFragment
+    }
+  }
+`;
+```
+
+### Step 2 — Map over the list in the component
+
+In the `Newsfeed` component, `data.topStories` will now be an array of fragment refs, each of which can be passed to a `Story` child component to render that story:
+
+```rescript
+@react.component
+let make = () => {
+  let data = NewsfeedQuery.use(~variables=(), ())
+  
+  switch topStories {
+  | None => React.null
+  | Some(stories) =>
+    <div className="newsfeed">
+      {stories
+      ->Belt.Array.keepMap(x => x)
+      ->Belt.Array.map(story => <Story story={story.fragmentRefs} />)
```

Comment:
```suggestion
      ->Array.keepSome
      ->Array.map(story => <Story story={story.fragmentRefs} />)
```
I suggest we use [Core](https://github.com/rescript-association/rescript-core) for all of the tutorial, if that's ok with you. We'd probably need to state that somewhere in the beginning. Other than that, it'll be fairly easy to adapt the functions used.

Reply:
Absolutely, I'm good with that!